### DISCUSSION TITLE
ANN: csv viewer

### DISCUSSION THREAD

**Discussion by williamstein** [sagemathinc/cocalc#6437] [Announcements]
In CoCalc, you can now view CSV files via a nice grid, rather than only having the option of looking at the underlying raw text.

```sh
git clone https://github.com/Opensourcefordatascience/Data-sets.git
cd Data-sets
open diamonds.csv
```

looks like this, and is showing a 3.1MB data table with over 50K rows:

<img width="1504" alt="image" src="https://user-images.githubusercontent.com/1276278/218868099-04abdca3-166c-4cfa-a6d4-cfe8ed86f6f0.png">

It should scroll smoothly despite the size of the data.

You can also split the pane and show the underlying raw data (which is editable):

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/1276278/218868364-ea0d3344-f8fa-4895-b239-a3ffd2bd0995.png">

The CSV grid viewer doesn't support collaborative editing yet, or publishing on the share server, but it still much nicer than just looking at raw text.

**Comment by DrXyzzy** [comment]
Online docs are updated to include this feature: https://doc.cocalc.com/csv-viewer.html
